### PR TITLE
Ensure GCD in `normalize_coordinates` for projective morphisms

### LINKED ISSUE #35797: normalize_coordinates for projective morphisms gcd failure
### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### Did you read the documentation and troubleshoot guide?

- [X] I have read the documentation and troubleshoot guide

### Environment

```markdown
- **OS**: Ubuntu 20.04
- **Sage Version**: 10.0
```

### Steps To Reproduce

```
K.<a> = NumberField(3*x^2 + 1)
P.<z,w> = ProjectiveSpace(K, 1)
phi = DynamicalSystem_projective([a*(z^2 + w^2),z*w])
phi.normalize_coordinates()
```

### Expected Behavior

Should return a version of the point with normalized coordinates

### Actual Behavior

the gcd step fails since it assumes clearing denominators gives back points in the ring of integers.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by guojing0**
### :books: Description

This PR also attempts to address and fixes #35797.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kwankyu** [comment]
3 doctests failed.

**Comment by bhutz** [comment]
Sorry, I'm still getting used to github, so I'm not I'm asking this correctly. This PR does not seem to be linked to the issue (#35797). Can you fix that?

**Comment by guojing0** [comment]
> Sorry, I'm still getting used to github, so I'm not I'm asking this correctly. This PR does not seem to be linked to the issue (#35797). Can you fix that?

I think in GitHub, the way to "link" to a specific issue is to mention or @ it, which I did in the description. So at the end of #35797, you can see "@[guojing0](https://github.com/guojing0) mentioned this issue Jun 22, 2023"

When this PR is merged eventually, we then can close that issue.

File: src/sage/schemes/projective/projective_morphism.py [hunk 1]
```diff
@@ -967,12 +979,20 @@ def normalize_coordinates(self, **kwds):
             self.scale_by(uniformizer**(-1 * min_val))
             return
 
-        # clear any denominators from the coefficients
+        R = self.domain().base_ring()
+        K = self.base_ring()
         N = self.codomain().ambient_space().dimension_relative() + 1
-        LCM = lcm([self[i].denominator() for i in range(N)])
```
Comment (bhutz) [inline src/sage/schemes/projective/projective_morphism.py:972]: There are still cases where you need to clear denominators like this (not in number fields). For example:

```
            sage: R.<a,b> = QQ[]
            sage: P.<x,y,z> = ProjectiveSpace(FractionField(R), 2)
            sage: H = End(P)
            sage: f = H([a/b*(x*z + y^2)*x^2, a*b*(x*z + y^2)*y^2, a*(x*z + y^2)*z^2])
            sage: f.normalize_coordinates(); f
```

File: src/sage/schemes/projective/projective_morphism.py [hunk 2]
```diff
@@ -967,12 +979,20 @@ def normalize_coordinates(self, **kwds):
...
 
-        R = self.domain().base_ring()
+        # Only clear denominators from the coefficients in the ring of integers
+        if R in NumberFields():
+            O = R.maximal_order()
+
+            if O is ZZ:
+                denom = lcm([self[i].denominator() for i in range(len(list(self)))])
+            else:
+                denom = R.ideal(list(self)).absolute_norm().denominator()
```
Comment (bhutz) [inline src/sage/schemes/projective/projective_morphism.py]: This is causing the errors. R is a number field and self is a list of polynomials, so this is not creating the right ideal. Something like the following should do what you want

```
denom = R.ideal([c for poly in self for c in poly.coefficients()]).absolute_norm().denominator()
```

**Comment by bhutz** [comment]
This is looking good.  I wasn't able to think of any examples where it didn't do what it claims to do. Looks like all tests passed the automated checks and my local tests.

Just one thing. In the new examples add the standard: 'checks that #35797 is fixed' in the docs.

**Comment by guojing0** [comment]
Done.

**Pull request merged**